### PR TITLE
fix: Fixes `Cannot find module 'any-promise'` on npm 2.x

### PR DESCRIPTION
**Pull request by kumar303**
I think this will fix https://github.com/mozilla-jetpack/jpm/pull/552 -- thanks to @rpl for help debugging!

### CODE REVIEW

**Comment by kumar303** [comment]
I checked that this failed with the old import statement. It's still kinda mind boggling as to why!! https://travis-ci.org/mozilla/sign-addon/jobs/144542658

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,11 @@ before_install:
 - git checkout master
 # Check out the commit that TravisCI started on:
 - git checkout -
-- npm install -g npm
+# On 0.12, we want to test with the default version of npm, otherwise
+# upgrade to the latest.
+- if [[ ${TRAVIS_NODE_VERSION:0:1} -gt "0" ]]; then
+  npm install -g npm;
```
Comment (rpl) [inline .travis.yml]: nit: can this line be indented 1-level inside the `if/fi`?

**Comment by rpl** [comment]
@kumar303 r+

I'm glad that we finally find a reasonable fix to supporto both npm 2.x and npm 3.x using the same package :-)

**Comment by kumar303** [comment]
I guess the webpack `require()` bootstrapper needs to enter the `mz/index.js` file (or something) but it's still really odd that you need npm 2.x to trigger the bug.

**Comment by kumar303** [comment]
upstream bugs:
* https://github.com/webpack/webpack/issues/2769
* https://phabricator.babeljs.io/T7491

**Pull request merged**